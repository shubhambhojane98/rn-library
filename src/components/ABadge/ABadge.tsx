import React, {FC} from 'react';
import {Text, StyleSheet} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';

interface Props {
  children: string | number;
  visible?: boolean;
  backgroundColor?: string;
  top?: number;
  left?: number;
  right?: number;
  bottom?: number;
  color?: string;
  borderRadius?: number;
  size?: number;
  theme: Theme;
}
const defaultSize = moderateScale(20, defaultScale);
const ABadge: FC<Props> = ({
  visible = true,
  children,
  backgroundColor,
  top,
  right,
  left,
  bottom,
  color,
  size = defaultSize,
  borderRadius = size / 2,
}) => {
  const {colors} = useTheme();
  const bgColor = backgroundColor ? backgroundColor : colors.primary;

  const stylesWithProp = styles({colors});

  return (
    <>
      {visible ? (
        <Text
          numberOfLines={1}
          style={{
            ...stylesWithProp.container,
            backgroundColor: bgColor,
            color,
            top,
            right,
            left,
            bottom,
            borderRadius,
            fontSize: size * 0.5,
            lineHeight: size,
            height: size,
            minWidth: size,
          }}>
          {children}
        </Text>
      ) : null}
    </>
  );
};

const styles = (props: {colors: any}) =>
  StyleSheet.create({
    container: {
      textAlign: 'center',
      textAlignVertical: 'center',
      paddingHorizontal: moderateScale(4, defaultScale),
      overflow: 'hidden',
      position: 'absolute',
    },
  });
export default withTheme(ABadge);
