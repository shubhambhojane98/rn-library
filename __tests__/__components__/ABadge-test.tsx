import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ABadge from '../../src/components/ABadge/ABadge';

it('renders correctly with label', () => {
  const tree = renderer.create(<ABadge children={10} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with visible false', () => {
  const tree = renderer
    .create(<ABadge children={10} visible={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
