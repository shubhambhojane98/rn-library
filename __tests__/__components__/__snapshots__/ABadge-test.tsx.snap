// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with label 1`] = `
<Text
  numberOfLines={1}
  style={
    Object {
      "backgroundColor": "#A81E23",
      "borderRadius": 12.285714285714285,
      "bottom": undefined,
      "color": undefined,
      "fontSize": 12.285714285714285,
      "height": 24.57142857142857,
      "left": undefined,
      "lineHeight": 24.57142857142857,
      "minWidth": 24.57142857142857,
      "overflow": "hidden",
      "paddingHorizontal": 4.914285714285715,
      "position": "absolute",
      "right": undefined,
      "textAlign": "center",
      "textAlignVertical": "center",
      "top": undefined,
    }
  }
>
  10
</Text>
`;

exports[`renders correctly with visible false 1`] = `null`;
